lib/linux_admin/network_interface.rb

```diff
@@ -0,0 +1,183 @@
+require 'ipaddr'
+
+module LinuxAdmin
+  class NetworkInterface
+    include Common
+
+    # Cached class instance variable for what distro we are running on
+    @dist_class = nil
+
+    # Gets the subclass specific to the local Linux distro
+    #
+    # @param clear_cache [Boolean] Determines if the cached value will be reevaluated
+    # @return [Class] The proper class to be used
+    def self.dist_class(clear_cache = false)
+      @dist_class = nil if clear_cache
+      @dist_class ||= begin
+        if [Distros.rhel, Distros.fedora].include?(Distros.local)
+          NetworkInterfaceRH
+        else
+          NetworkInterfaceGeneric
+        end
+      end
+    end
+
+    class << self
+      private
+
+      alias_method :orig_new, :new
+    end
+
+    # Creates an instance of the correct NetworkInterface subclass for the local distro
+    def self.new(*args)
+      if self == LinuxAdmin::NetworkInterface
+        dist_class.new(*args)
+      else
+        orig_new(*args)
+      end
+    end
+
+    # @return [String] the interface for networking operations
+    attr_reader :interface
+
+    # @param interface [String] Name of the network interface to manage
+    def initialize(interface)
+      @interface = interface
+      reload
+    end
+
+    # Gathers current network information for this interface
+    def reload
+      @network_conf = {}
+      return false unless (ip_output = ip_show)
+
+      parse_ip4(ip_output)
+      parse_ip6(ip_output, :global)
+      parse_ip6(ip_output, :link)
+
+      @network_conf[:mac] = parse_ip_output(ip_output, %r{link/ether}, 1)
+
+      ip_route_res = run(cmd("ip"), :params => ["route"])
+      @network_conf[:gateway] = parse_ip_output(ip_route_res.output, /^default/, 2) if ip_route_res.success?
+      true
+    end
+
+    # Retrieve the IPv4 address assigned to the interface
+    #
+    # @return [String] IPv4 address for the managed interface
+    def address
+      @network_conf[:address]
+    end
+
+    # Retrieve the IPv6 address assigned to the interface
+    #
+    # @return [String] IPv6 address for the managed interface
+    # @raise [ArgumentError] if the given scope is not `:global` or `:link`
+    def address6(scope = :global)
+      if scope == :global
+        @network_conf[:address6_global]
+      elsif scope == :link
+        @network_conf[:address6_link]
+      else
+        raise ArgumentError, "Unrecognized address scope #{scope}"
+      end
+    end
+
+    # Retrieve the MAC address associated with the interface
+    #
+    # @return [String] the MAC address
+    def mac_address
+      @network_conf[:mac]
+    end
+
+    # Retrieve the IPv4 sub-net mask assigned to the interface
+    #
+    # @return [String] IPv4 netmask
+    def netmask
```

Comment:
I'm fine with any other than just mask.  I think it is pretty intuitive either way, the terms netmask and subnet mask are used pretty interchangeably.